code/modules/library/computers/remote_gallery_access.dm

```diff
@@ -0,0 +1,113 @@
+			<A href='?src=\ref[src];settitle=1'>Filter by Title: [query.title]</A><br />
+			<A href='?src=\ref[src];setcategory=1'>Filter by Category: [query.category]</A><br />
+			<A href='?src=\ref[src];setauthor=1'>Filter by Author: [query.author]</A><br />
+			<A href='?src=\ref[src];search=1'>\[Start Search\]</A><br />"}
+		dat += pagelist
+
+		dat += {"<table border=\"0\">
+					<tr>
+						<td>Author</td>
+						<td>Title</td>
+						<td>Painting</td>
+						<td>Description</td>
+						<td>Controls</td>
+					</tr>"}
+
+		for(var/datum/cachedbook/C in get_page(page_num))
+			var/author = C.author
+			var/datum/custom_painting/the_painting = json2painting(C.content)
+			var/controls =  "<A href='?src=\ref[src];id=[C.id]'>\[Order\]</A>"
+			if(isAdminGhost(user))
+				author += " (<A style='color:red' href='?src=\ref[src];delbyckey=[ckey(C.ckey)]'>[ckey(C.ckey)])</A>)"
+			if(isAdminGhost(user) || allowed(user))
+				controls +=  " <A style='color:red' href='?src=\ref[src];del=[C.id]'>\[Delete\]</A>"
+			dat += {"<tr>
+						<td>[author]</td>
+						<td>[C.title]</td>
+						<td><img src='data:image/png;base64,[icon2base64(the_painting.render_on(icon('icons/obj/paintings.dmi', "blank")))]'></td>
+						<td>[C.description]</td>
+						<td>
+							[controls]
+						</td>
+					</tr>"}
+
+		dat += "</table><br />[pagelist]"
+		dat += "<hr><br/> <a href='?src=\ref[src];upload=1'>Upload new painting</a>"
+
+	var/datum/browser/B = new /datum/browser/clean(user, "remote_gallery", "Remote Gallery Viewer")
+	B.set_content(dat)
+	B.open()
+
+/obj/machinery/computer/library/checkout/remote_gallery/get_scanner_title(var/obj/machinery/libraryscanner/LS)
+	return LS.cached_painting.painting_data.title
+
+/obj/machinery/computer/library/checkout/remote_gallery/get_scanner_author(var/obj/machinery/libraryscanner/LS)
+	return LS.cached_painting.painting_data.author
+
+/obj/machinery/computer/library/checkout/remote_gallery/get_scanner_dat(var/obj/machinery/libraryscanner/LS)
+	return painting2json(LS.cached_painting.painting_data)
+
+/obj/machinery/computer/library/checkout/remote_gallery/get_scanner_category(var/obj/machinery/libraryscanner/LS)
+	return "[LS.cached_painting.painting_height]x[LS.cached_painting.painting_width]"
+
+/obj/machinery/computer/library/checkout/remote_gallery/get_scanner_desc(var/obj/machinery/libraryscanner/LS)
+	return LS.cached_painting.painting_data.description
+
+/obj/machinery/computer/library/checkout/remote_gallery/has_cached_data()
+	return scanner.cached_painting
+
+/obj/machinery/computer/library/checkout/remote_gallery/make_external_book(var/datum/cachedbook/newbook)
+	message_admins("printing [newbook.title]...")
+	if(!newbook)
+		return
+	var/obj/item/mounted/frame/painting/custom/C = new(get_turf(src))
+	C.name = "[newbook.title] by [newbook.author]"
+	C.desc = newbook.description
+	C.painting_data = json2painting(newbook.content, newbook.title, newbook.author, newbook.description)
```

Comment:
```suggestion
	C.set_painting_data(json2painting(newbook.content, newbook.title, newbook.author, newbook.description))
```
Doing it this way will ensure `C.painting_data.set_parent(C)` gets called and the multipart handler gets set. 
...Don't remember what exactly would happen if it's not set, probably just being unable to paint until a player hangs the painting (likely the first thing most players will do).
So nothing serious, but still.